components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3127,14 +3127,23 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String
 
     public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2Exception {
 
-        IdentityProvider identityProvider = getResidentIdp(tenantDomain);
-        FederatedAuthenticatorConfig[] fedAuthnConfigs = identityProvider.getFederatedAuthenticatorConfigs();
-        // Get OIDC authenticator
-        FederatedAuthenticatorConfig oidcAuthenticatorConfig =
-                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
-                        IdentityApplicationConstants.Authenticator.OIDC.NAME);
-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
-                IDP_ENTITY_ID).getValue();
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();
+            } catch (URLBuilderException e) {
+                String errorMsg = String.format("Error while building url for '%s' context", OAUTH2_TOKEN_EP_URL);
```

Comment:
I think it would be better say "Error while retrieving ID Token issuer value for tenantDomain: " + tenantDomain
here. 

Can you check how the exception thrown is handled? If we are handling and logging a similar message as I suggested above we do not need to change.

Reply:
It only has "Error while retrieving OIDC Id token issuer" + e. No mention of a tenant domain. Should we add the tenant domain there?

Reply:
Okay let add it there

Reply:
Also lets try to include the tenantDomain herr too

Reply:
Updated the PR with the change